claasp/ciphers/block_ciphers/ublock_block_cipher.py

```diff
@@ -0,0 +1,335 @@
+
+# ****************************************************************************
+# Copyright 2023 Technology Innovation Institute
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# ****************************************************************************
+
+
+from claasp.cipher import Cipher
+from claasp.DTOs.component_state import ComponentState
+from claasp.name_mappings import INPUT_PLAINTEXT, INPUT_KEY
+
+PARAMETERS_CONFIGURATION_LIST = [
+    {'block_bit_size': 128, 'key_bit_size': 128, 'r': 16},
+    {'block_bit_size': 128, 'key_bit_size': 256, 'r': 24},
+    {'block_bit_size': 256, 'key_bit_size': 256, 'r': 24},
+]
+
+SBOX = [0x7, 0x4, 0x9, 0xc, 0xb, 0xa, 0xd, 0x8, 0xf, 0xe, 0x1, 0x6, 0x0, 0x3, 0x2, 0x5]
+SBOX_TK = [0x0, 0x2, 0x4, 0x6, 0x8, 0xa, 0xc, 0xe, 0x3, 0x1, 0x7, 0x5, 0xb, 0x9, 0xf, 0xd]
+SBOX_SIZE = 4
+PL = [
+    [1, 3, 4, 6, 0, 2, 7, 5],
+    [2, 7, 8, 13, 3, 6, 9, 12, 1, 4, 15, 10, 14, 11, 5, 0]
+]
+PR = [
+    [2, 7, 5, 0, 1, 6, 4, 3],
+    [6, 11, 1, 12, 9, 4, 2, 15, 7, 0, 13, 10, 14, 3, 8, 5]
+]
+P_WORD_SIZE = 8
+PK = [
+    [6, 0, 8, 13, 1, 15, 5, 10, 4, 9, 12, 2, 11, 3, 7, 14],
+    [10, 5, 15, 0, 2, 7, 8, 13, 14, 6, 4, 12, 1, 3, 11, 9, 24, 25, 26, 27, 28, 29, 30, 31, 16, 17, 18, 19, 20, 21, 22,
+     23],
+    [10, 5, 15, 0, 2, 7, 8, 13, 1, 14, 4, 12, 9, 11, 3, 6, 24, 25, 26, 27, 28, 29, 30, 31, 16, 17, 18, 19, 20, 21, 22,
+     23]
+]
+PK_WORD_SIZE = 4
+RC = [
+    0x988cc9dd, 0xf0e4a1b5, 0x21357064, 0x8397d2c6, 0xc7d39682, 0x4f5b1e0a, 0x5e4a0f1b, 0x7c682d39,
+    0x392d687c, 0xb3a7e2f6, 0xa7b3f6e2, 0x8e9adfcb, 0xdcc88d99, 0x786c293d, 0x30246175, 0xa1b5f0e4,
+    0x8296d3c7, 0xc5d19480, 0x4a5e1b0f, 0x55410410, 0x6b7f3a2e, 0x17034652, 0xeffbbeaa, 0x1f0b4e5a,
+]
+RC_SIZE = 32
+ROTATE_SIZE = 32
+
+class UblockBlockCipher(Cipher):
+    """
+    Construct an instance of the UblockBlockCipher class.
+
+    This class is used to store compact representations of a cipher, used to generate the corresponding cipher.
```

Comment:
This sentence can be removed. Instead, add a sentence with a reference to the technical specifications and the test vectors.